### PR TITLE
LLM Data Leakage Experiment

### PR DESCRIPTION
**Pull request by priyanka-operant**
v1 response:
```
{
    "k8s_version": "v1.23.17-eks-3af4770",
    "results": [
        {
            "experiment": "llm-data-leakage",
            "description": "This experiment checks whether the LLM AI Model is leaking any sensitive data such as PII data or secrets and keys in its response",
            "framework": "MITRE-ATLAS",
            "tactic": "Exfiltration",
            "technique": "LLM Data Leakage",
            "result": {
                "Check for PII data leakage in the AI model response": "fail"
            }
        }
    ]
}
```
Todo: Add more info in the output about why the experiment failed

### CODE REVIEW

File: internal/experiments/experiments_ai_data_leakage.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+	Status         int           `json:"status"`
+	Response       AIAPIResponse `json:"response"`
+}
+
+func (p *LLMDataLeakageExperiment) Type() string {
+	return "llm_data_leakage"
+}
+
+func (p *LLMDataLeakageExperiment) Description() string {
+	return "This experiment checks whether the LLM AI Model is leaking any sensitive data such as PII data or secrets and keys in its response"
```
Comment (glenn-operant) [inline internal/experiments/experiments_ai_data_leakage.go:66]: I've been trying to keep this shorter maybe:

```
Checks whether the LLM AI Model is leaking any sensitive data such as PII data or secrets and keys in its response
```

File: internal/experiments/experiments_ai_data_leakage.go [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+	return categories.MITREATLAS.Exfiltration.LLMDataLeakage.Technique
+}
+func (p *LLMDataLeakageExperiment) Tactic() string {
+	return categories.MITREATLAS.Exfiltration.LLMDataLeakage.Tactic
+}
+func (p *LLMDataLeakageExperiment) Framework() string {
+	return string(categories.MitreAtlas)
+}
+
+func (p *LLMDataLeakageExperiment) Run(ctx context.Context, client *k8s.Client, experimentConfig *ExperimentConfig) error {
```
Comment (glenn-operant) [inline internal/experiments/experiments_ai_data_leakage.go:78]: Should we add some guards here to ensure that the secops-chaos-ai component is deployed?

File: internal/experiments/experiments_ai_data_leakage.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+type ExecuteAIAPIResult struct {
+	ExperimentName string        `json:"experiment_name"`
+	Description    string        `json:"description"`
+	Timestamp      time.Time     `json:"timestamp"`
+	Status         int           `json:"status"`
+	Response       AIAPIResponse `json:"response"`
+}
+
+func (p *LLMDataLeakageExperiment) Type() string {
+	return "llm_data_leakage"
```
Comment (glenn-operant) [inline internal/experiments/experiments_ai_data_leakage.go:62]:
```suggestion
	return "llm-data-leakage"
```

Been changing these to dashes for consistency

File: experiments/llm_data_leakage.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+experiments:
```
Comment (glenn-operant) [inline experiments/llm_data_leakage.yaml:1]: Rename to llm-data-leakage.yaml to match the rest of the experiments now due to "snippets" change

**Review by glenn-operant — COMMENTED** [review]
There's a Experiments [README](https://github.com/OperantAI/secops-chaos/blob/main/experiments/README.md) that maintains a list of experiments, and their descriptions. Could you add this to it?

**Comment by glenn-operant** [comment]
Rebased this PR to make it's build pass due to GHA changes I had done.

**Review by glenn-operant — APPROVED** [review]

**Comment by priyanka-operant** [comment]
Closes #63

**Pull request merged**